pysat/tests/test_utils_files.py

```diff
@@ -296,3 +298,120 @@ def test_updating_directories(self, capsys):
+        return
+
+    def test_get_file_information(self):
+        """Test `utils.files.get_file_information` success with existing files.
+
+        """
+
+        # Create a bunch of files by year and doy
+        root_fname = ''.join(('pysat_testing_junk_{year:04d}_gold_',
+                              '{day:03d}_stuff.pysat_testing_file'))
+        pimtesting.create_files(self.testInst, self.start, self.stop, freq='1D',
+                                root_fname=root_fname)
+
+        # Use from_os function to get pandas Series of files and dates
+        files = pysat.Files.from_os(data_path=self.testInst.files.data_path,
+                                    format_str=root_fname)
+
+        # Get file attributes
+        root_dir = self.testInst.files.data_path
+        file_info = futils.get_file_information(files.values,
+                                                root_dir=root_dir)
+
+        # Check overall length for each output key.
+        for key in file_info.keys():
+            assert len(file_info[key]) == len(files)
+
+        # Ensure general correctness of time. If epoch or units wrong
+        # then time will be way off. Allowing for difference between local
+        # and UTC times.
+        today = pysat.utils.time.today()
+        assert np.all(file_info['content_modified_time']
+                      <= today + dt.timedelta(days=1))
+        assert np.all(file_info['content_modified_time']
+                      >= today - dt.timedelta(days=1))
+
+        return
+
+    def test_check_and_make_path_exists(self):
+        """Test successful pass at creating existing directory."""
+
+        # Create a temporary directory
+        tempdir = tempfile.TemporaryDirectory()
+        assert os.path.isdir(tempdir.name)
+
+        # Assert prep_dir does not re-create the directory
+        assert not pysat.utils.files.check_and_make_path(tempdir.name)
+
+        # Clean up temporary directory
+        tempdir.cleanup()
+        return
+
+    @pytest.mark.parametrize("trailer", [None, '', 'extra',
+                                         os.path.join('extra', 'extra')])
+    def test_check_and_make_path_new(self, trailer):
+        """Test successful pass at creating existing directory."""
+
+        # Create a temporary directory and get its name
+        tempdir = tempfile.TemporaryDirectory()
+        new_dir = tempdir.name
+
+        if trailer is not None:
+            new_dir = os.path.join(new_dir, trailer)
+
+        # Clean up temporary directory
+        tempdir.cleanup()
+        assert not os.path.isdir(new_dir)
+
+        # Assert prep_dir re-creates the directory
+        assert pysat.utils.files.check_and_make_path(new_dir)
+
+        # Clean up the test directory
+        os.rmdir(new_dir)
+        return
+
+    @pytest.mark.parametrize("path", ['no_starting_path_info',
+                                      os.path.join('no', ' way ', ' brah ')])
+    def test_check_and_make_path_error(self, path):
```

Comment:
Also need a ValueError test for 'Desired and constructed paths differ'.

Reply:
Right.... 🤔

Reply:
If it's impossible to reach that error catch we can get rid of it.  Part of my C background has me paranoid about the possibilities of corrupted strings/memory.  So I may have added a completely unnecessary test!

Reply:
I poked at it a few times but wasn't able to get there. 

> Part of my C background has me paranoid about the possibilities of corrupted strings/memory.  So I may have added a completely unnecessary test

No worries! When it comes to changing a file system I'm all for safety checks. I expanded the text to ask users to post an issue if they ever hit that error.